lib/Dist/Zilla/Plugin/InsertExample.pm

```diff
@@ -119,17 +137,34 @@ and it won't be a verbatim paragraph at all.
 
     my $indent = ' ' x $self->indent;
 
+    my $in_boiler = 1;
+    my $found_content = 0;
     while(my $line = <$fh>)
     {
       if($self->remove_boiler)
       {
-        next if $line =~ /^\s*$/;
-        next if $line =~ /^#!\/usr\/bin\/perl/;
-        next if $line =~ /^#!\/usr\/bin\/env perl/;
-        next if $line =~ /^use strict;$/;
-        next if $line =~ /^use warnings;$/;
+          if($self->has_match_boiler_barrier)
+          {
+              if($in_boiler)
+              {
+                  $in_boiler = 0 if $line =~ $self->match_boiler_barrier;
+                  next;
+              }
+          }
+          else
+          {
+              next if $line =~ /^\s*$/;
+              next if $line =~ /^#!\/usr\/bin\/perl/;
+              next if $line =~ /^#!\/usr\/bin\/env perl/;
+              next if $line =~ /^use strict;$/;
+              next if $line =~ /^use warnings;$/;
```

Comment:
Could these be the default for `match_boiler_barrier` instead of coding them here?